dataflow/src/ops/rewrite.rs

```diff
@@ -100,8 +100,12 @@ impl Ingredient for Rewrite {
 
 
                 if rc.is_none() {
-                    // todo (larat): might need to do some stuff here
-                    continue;
+                    misses.push(Miss {
+                       node: *self.signal,
+                       columns: vec![0],
+                       replay_key: None,
```

Comment:
I *think* `replay_key` needs to be set here too.

Reply:
Uhmm, so the `signal` parent is just used to check if the record should be rewritten or not, none of its columns actually make it into the output of the node (all columns come from the `src` parent).

Reply:
But this is under `if from == *self.src`, right? Think of it this way: if we miss doing a lookup into `signal`, we'll want to first replay the key we missed on *to* `signal`, and then replay the key we were working on when we missed.

Reply:
I'm still a little confused about this. Would this mean that under `if from == *self.signal`, a miss in `src` would have a `None` `replay_key`? 

Otherwise, it seems weird that in both cases you can set `replay_key` to `replay_key_col.map(|col| vec![r[col].clone()])`, given what you said about `replay_key` being the column in the input that corresponds to the `replay_key_col` column in the output.

Reply:
I think the situation is as follows: if we receive a record on `src`, and that record has a `key` value that we've never seen before (e.g., a Post ID), the partially-materialized `signal` parent will not have data for that key (the Post ID) and will need to query backwards up the graph to find out whether to rewrite the record or not. Hence, it needs to initiate a replay keyed on the Post ID that the corresponding column in the `src` input record holds.

Reply:
Uhm, but then, how is `replay_key` different from `key` (which is also a parameter in `Miss)?

Reply:
`key` is the key you used in the call to `lookup`. `replay_key` is the value of `replay_key_col` in the record you were processing when the lookup missed.

Reply:
With the slight caveat that `replay_key_col` refers to an *output* column, so you have to trace it back to the input.